### PR TITLE
Update preflight validations for Nutanix

### PR DESCRIPTION
**Pull request by thunderboltsid**
* Add new preflight validations for machine specs, endpoint, port, and credentials
* Refactor the unit tests for validations into table tests

### CODE REVIEW

File: pkg/providers/nutanix/mock_validator_test.go [hunk 1]
```diff
@@ -1,48 +0,0 @@
-// Code generated by MockGen. DO NOT EDIT.
```
Comment (thunderboltsid) [inline pkg/providers/nutanix/mock_validator_test.go:1]: This file seems like a stray. We are using the mocks from crypto/mocks instead.

**Comment by deepakm-ntnx** [comment]
Adds following point from https://github.com/aws/eks-anywhere/issues/4167
Add preflight to validate connection to NutanixDatacenterConfig's endpoint:port using something similar to this
[eks-anywhere/pkg/executables/govc.go](https://github.com/aws/eks-anywhere/blob/2a03ef2bb651cfc70c1cfea12a032103263d7817/pkg/executables/govc.go#L598-L604)

Lines 598 to 604 in [2a03ef2](https://github.com/aws/eks-anywhere/commit/2a03ef2bb651cfc70c1cfea12a032103263d7817)

 skipVerifyTransport := http.DefaultTransport.(*http.Transport).Clone() 
 skipVerifyTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true} 
 client := &http.Client{Transport: skipVerifyTransport} 
  
 if _, err := client.Get("https://" + server); err != nil { 
 	return fmt.Errorf("failed to reach server %s: %v", server, err) 
 } 
 Add preflight to validate Nutanix credentials before validating subnets and images
 Add preflight to validate NutanixDatacenterConfig.endpoint is a valid endpoint
 Add preflight to validate NutanixDatacenterConfig.port is a valid port
 Add preflight to validate NutanixDatacenterConfig.additionalTrustBundle, if provided, is a valid PEM encoded cert
 Add preflight to validate NutanixMachineConfig's memorySize, systemDiskSize, vcpuSockets and vcpusPerSocket
 Separate out validation errors for cluster/subnet/image instead of wrapping them https://github.com/aws/eks-anywhere/blob/main/pkg/providers/nutanix/validator.go#L45-L55

File: pkg/providers/nutanix/validator.go [hunk 2]
```diff
@@ -33,32 +45,108 @@ func (v *Validator) ValidateDatacenterConfig(ctx context.Context, config *anywhe
...
+	if err := v.validateTrustBundleConfig(config.Spec); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (v *Validator) validateEndpointAndPort(dcConf anywherev1.NutanixDatacenterConfigSpec) error {
+	if !networkutils.IsPortValid(strconv.Itoa(dcConf.Port)) {
+		return fmt.Errorf("nutanix prism central port %q out of range", dcConf.Port)
```
Comment (abhinavmpandey08) [inline pkg/providers/nutanix/validator.go:66]:
```suggestion
		return fmt.Errorf("nutanix prism central port %d out of range", dcConf.Port)
```

Comment (deepakm-ntnx) [reply pkg/providers/nutanix/validator.go]: resolved.

**Review by abhinavmpandey08 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**